src/scry/implementations.cr

```diff
@@ -49,15 +69,19 @@ module Scry
     private def analyze(filename, position, scope)
       response = crystal_tool(filename, position, scope)
       Log.logger.debug("response: #{response}")
-      parsed_response = JSON.parse(response)
-      implementations = parsed_response["implementations"]?
-      if implementations
-        implementations
-      else
+      response = (Array(BuildFailure) | ImplementationsResponse).from_json(response)
+      case response
+      when Array(BuildFailure)
         implementation_response
+      when ImplementationsResponse
+        if impls = response.implementations
+          impls
```

Comment:
why not always return a response by doing `response_with(impls)` here?
(and remove the case in `search`)

Reply:
Done!